cmd/ame/cli_test.go

```diff
@@ -42,51 +45,114 @@ var (
 	ctx       context.Context
 )
 
-func kubeClientFromConfig() (*rest.Config, error) {
-	configLoadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	configOverrides := &clientcmd.ConfigOverrides{}
-	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(configLoadingRules, configOverrides)
-	config, err := kubeConfig.ClientConfig()
-	if err != nil {
-		return nil, err
-	}
+func waitForWorkflowStatus(ctx context.Context, wfName string, timeout time.Duration, status argoWf.WorkflowPhase) error {
+	deadline := time.Now().Add(timeout)
 
-	return config, nil
-}
+	var wf *argoWf.Workflow
+	for {
+		if time.Now().After(deadline) {
+			wfStatus := ""
+
+			if wf != nil {
+				wfStatus = string(wf.Status.Phase)
+			}
+			return fmt.Errorf("workflow: %s did not reach the target phase: %s within: %v, ended at phase: %s", wfName, status, timeout, wfStatus)
+		}
+		var err error
+		wf, err = workflows.Get(ctx, wfName, v1.GetOptions{})
+		if err != nil {
+			return err
+		}
 
-func workflowsClientFromConfig(cfg *rest.Config, ns string) argo.WorkflowInterface {
-	return argo.NewForConfigOrDie(cfg).Workflows(ns)
+		if wf.Status.Phase == status {
+			return nil
+		}
+
+		time.Sleep(time.Millisecond * 100)
+	}
 }
 
-func tasksClientFromConfig(cfg *rest.Config, ns string) v1alpha1.TaskInterface {
-	return v1alpha1.NewForConfigOrDie(cfg).Tasks(ns)
+func getWorkflowForProject(ctx context.Context, projectId string, timeout time.Duration) (*argoWf.Workflow, error) {
```

Comment:
I think this belongs in a package under /internal for general use within the project. Perhaps in the ameproject package?